clean/new_orleans_so_cprr.py

```diff
@@ -113,14 +273,222 @@ def split_investigating_supervisor(df):
         .str.extract(r'^([^ ]+) (.+)')
     df.loc[:, 'supervisor_first_name'] = names[0]
     df.loc[:, 'supervisor_last_name'] = names[1]
-    return df.drop(columns=['investigating_supervisor'])
+    return df.drop(columns=['investigating_supervisor', 'supervisor_name'])
+
+
+def clean_receive_date_20(df):
+    df.loc[:, 'receive_date'] = df.date_received.fillna('')\
+        .str.replace(r'(\d+)/(\d+)//(\d+)', r'\1/\2/\3', regex=True)\
+        .str.replace('!2', '12', regex=False)\
+        .str.replace('9/142020', '9/14/2020', regex=False)
+    return df.drop(columns='date_received')
+
+
+def clean_investigation_start_date_20(df):
+    df.loc[:, 'investigation_start_date'] = df.date_started.fillna('')\
+        .str.replace(r'(\d+)//(\d+)$', r'\1/\2', regex=True)
+    return df.drop(columns='date_started')
+
+
+def clean_investigation_complete_date_20(df):
+    df.loc[:, 'investigation_complete_date'] = df.date_completed.fillna('')\
+        .str.replace(r'^(\d+)//(\d+)', r'\1/\2', regex=True)\
+        .str.replace('031/2/2020', '3/12/2020', regex=False)\
+        .str.replace('3/42021', '3/4/2021', regex=False)
+    return df.drop(columns='date_completed')
+
+
+def clean_action_20(df):
+    df.action = df.action.str.lower().str.strip()\
```

Comment:
Please use `.loc` here. Maybe we can use an automated linter for these things.

Reply:
I will look into it.